pyspark/graphar_pyspark/graph.py

```diff
@@ -0,0 +1,215 @@
+    ) -> None:
+        """Write graph data in graphar format.
+
+
+        Note: for default parameters check com.alibaba.graphar.GeneralParams;
+        For this method None for any of arguments means that the default value will be used.
+
+        :param path: the directory to write.
+        :param name: the name of graph, default is 'grpah'
+        :param vertex_chunk_size: the chunk size for vertices, default is 2^18
+        :param edge_chunk_size: the chunk size for edges, default is 2^22
+        :param file_type: the file type for data payload file, support [parquet, orc, csv], default is parquet.
+        :param version: version of graphar format, default is v1.
+        """
+        if vertex_chunk_size is None:
+            vertex_chunk_size = (
+                GraphArSession._graphar.GeneralParams.defaultVertexChunkSize
+            )
+
+        if edge_chunk_size is None:
+            edge_chunk_size = GraphArSession._graphar.GeneralParams.defaultEdgeChunkSize
+
+        if file_type is None:
+            file_type = GraphArSession._graphar.GeneralParams.defaultFileType
+        else:
+            file_type = file_type.value
+
+        if version is None:
+            version = GraphArSession._graphar.GeneralParams.defaultVersion
+
+        self._jvm_graph_writer_obj.write(
+            path,
+            GraphArSession._jss,
+            name,
+            vertex_chunk_size,
+            edge_chunk_size,
+            file_type,
+            version,
+        )
+
+
+class GraphTransformer:
+    @staticmethod
+    def transform(
+        source_graph_info: Union[str, GraphInfo], dest_graph_info: Union[str, GraphInfo]
+    ) -> None:
+        """Transform the graphs following the meta data provided or defined in info files.
+
+        Note: both arguments should strings or GrapInfo instances! Mixed arguments type is not supported.
+
+        :param source_graph_info: The path of the graph info yaml file for the source graph OR the info object for the source graph.
+        :param dest_graph_info: The path of the graph info yaml file for the destination graph OR the info object for the destination graph.
+        """
+        if isinstance(source_graph_info, str) and isinstance(dest_graph_info, str):
+            GraphArSession._graphar.graph.GraphTransformer.transform(
+                source_graph_info, dest_graph_info, GraphArSession._jss
+            )
+        elif isinstance(source_graph_info, GraphInfo) and isinstance(
+            dest_graph_info, GraphInfo
+        ):
+            GraphArSession._graphar.graph.GraphTransformer.transform(
+                source_graph_info.to_scala(),
+                dest_graph_info.to_scala(),
+                GraphArSession._jss,
+            )
+        else:
+            msg = "Both src and dst graph info objects should be of the same type. "
+            msg += f"But {type(source_graph_info)} and {type(dest_graph_info)} were provided!"
+            raise ValueError(msg)
```

Comment:
Minor: I would be great to document the exception type in the docstring (`:raises ...`), and it also would be super to define a custom exception (something like `class InvalidGraphTransformInfos(ValueError):...`).

Reply:
Done